src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java

```diff
@@ -206,4 +205,24 @@ private static Reference[] convertReferences(List<String> xrayReferences) {
             return new Reference(parts[1].substring(0, parts[1].length() - 1), parts[0].substring(1));
         }).toArray(Reference[]::new);
     }
+
+    private static String getPackageTypeName(String componentId) {
+        String GENERIC_PKG_TYPE = "Generic";
+        String[] compIdParts = componentId.split("://");
+        if (compIdParts.length != 2) {
+            return GENERIC_PKG_TYPE;
+        }
+        switch (compIdParts[0]) {
```

Comment:
You can switch here, somehow on ComponentPrefix.valueOf(...) (requires some changes)